### DISCUSSION TITLE
HeteroData import question

### DISCUSSION THREAD

**Discussion by danielpatrickhug** [pyg-team/pytorch_geometric#3094] [Q&A] [answered]
Hi, I noticed that the line given in a lot of tutorials - [ex1](https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html), [ex2](https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html#example-graph)
`from torch_geometric.data import HeteroData`
returns the error in colab

```
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-ae3d725bb52c> in <module>()
----> 1 from torch_geometric.data import HeteroData
*Updated
>ImportError: cannot import name 'HeteroData' from 'torch_geometric.data'

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
```
After looking through the torch_geometric.data object dict
`torch_geometric.data.__dict__`

```
{'Batch': torch_geometric.data.batch.Batch,
 'ClusterData': <function torch_geometric.loader.cluster.ClusterData>,
 'ClusterLoader': <function torch_geometric.loader.cluster.ClusterLoader>,
 'Data': torch_geometric.data.data.Data,
 'DataListLoader': <function torch_geometric.loader.data_list_loader.DataListLoader>,
 'DataLoader': <function torch_geometric.loader.dataloader.DataLoader>,
 'Dataset': torch_geometric.data.dataset.Dataset,
 'DenseDataLoader': <function torch_geometric.loader.dense_data_loader.DenseDataLoader>,
 'GraphSAINTEdgeSampler': <function torch_geometric.loader.graph_saint.GraphSAINTEdgeSampler>,
 'GraphSAINTNodeSampler': <function torch_geometric.loader.graph_saint.GraphSAINTNodeSampler>,
 'GraphSAINTRandomWalkSampler': <function torch_geometric.loader.graph_saint.GraphSAINTRandomWalkSampler>,
 'GraphSAINTSampler': <function torch_geometric.loader.graph_saint.GraphSAINTSampler>,
 'HeteroData': torch_geometric.data.hetero_data.HeteroData,
 'InMemoryDataset': torch_geometric.data.in_memory_dataset.InMemoryDataset,
 'NeighborSampler': <function torch_geometric.loader.neighbor_sampler.NeighborSampler>,
 'RandomNodeSampler': <function torch_geometric.loader.random_node_sampler.RandomNodeSampler>,
 'ShaDowKHopSampler': <function torch_geometric.loader.shadow.ShaDowKHopSampler>,
 'TemporalData': torch_geometric.data.temporal.TemporalData,
...
```
I was able to import HeteroData using -
`from torch_geometric.data.hetero_data import HeteroData`

Is this the correct import and would I be correct to make a pull request replacing the former with the latter?

**Comment by rusty1s** [comment, marked as answer]
`from torch_geometric.data import HeteroData` should work out-of-the-box. Can you point me to the Colab to have a look?

**Reply by danielpatrickhug** [reply]
Thanks for responding, I'm no longer receiving that error. Here is the [notebook](https://colab.research.google.com/drive/1EzmetZjCBOhi-MzwPy87ONOj-idMmBKo?usp=sharing) in question

**Reply by danielpatrickhug** [reply]
I was going through the tutorial code

**Reply by rusty1s** [reply]
Thanks. Do you know what caused this error and why it is now resolved?

**Reply by danielpatrickhug** [reply]
I don't, I'm confident I installed the packages before I imported. I tried this 3 days ago and got an error - this is from my local box but I got the same one from colab. 
I meant to send this error
```
    from torch_geometric.data import HeteroData
ImportError: cannot import name 'HeteroData' from 'torch_geometric.data' (/usr/local/anaconda3/envs/pytorch-geo/lib/python3.9/site-packages/torch_geometric/data/__init__.py)
```

**Reply by danielpatrickhug** [reply]
That lead me to look at the object __dict__

**Reply by maximocrv** [reply]
I am currently having the same issue, I installed PyG via conda (using the channels in the readme). I checked in the `data` directory and the `hetero_data.py` file is not in there. Edit: just saw that it is not officially released and needs to be installed via pip.

**Reply by rusty1s** [reply]
Yes, we plan to release this by next monday when we finished fixing all the bugs :)
